### ISSUE TITLE
"ValueError: no path specified" when generating ninja build files

### ISSUE THREAD

**Issue by FSMaxB** [mesonbuild/meson#2011]
I recently updated to meson 0.41.1 and meson stopped producing ninja build files with the following error:

```
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/mesonbuild/mesonmain.py", line 307, in run
    app.generate()
  File "/usr/lib/python3.6/site-packages/mesonbuild/mesonmain.py", line 182, in generate
    g.generate(intr)
  File "/usr/lib/python3.6/site-packages/mesonbuild/backend/ninjabackend.py", line 194, in generate
    self.generate_target(t, outfile)
  File "/usr/lib/python3.6/site-packages/mesonbuild/backend/ninjabackend.py", line 445, in generate_target
    elem = self.generate_link(target, outfile, outname, obj_list, linker, pch_objects)
  File "/usr/lib/python3.6/site-packages/mesonbuild/backend/ninjabackend.py", line 2331, in generate_link
    target.install_rpath)
  File "/usr/lib/python3.6/site-packages/mesonbuild/compilers.py", line 839, in build_rpath_args
    return self.build_unix_rpath_args(build_dir, from_dir, rpath_paths, install_rpath)
  File "/usr/lib/python3.6/site-packages/mesonbuild/compilers.py", line 768, in build_unix_rpath_args
    relative = os.path.relpath(p, from_dir)
  File "/usr/lib/python3.6/posixpath.py", line 448, in relpath
    raise ValueError("no path specified")
ValueError: no path specified
```

* Meson-Version meson 0.41.1
* System: Archlinux
* Python: 3.6.1

Here is my project so you can reproduce the issue: https://github.com/DaveGamble/cJSON/tree/meson-issue

**Comment by QuLogic** [comment]
This is fixed by #1926.

**Comment by FSMaxB** [comment]
Oh, I was only searching for existing issues, not pull requests.

I can confirm that #1926 fixes the issue.
